### DISCUSSION TITLE
Can I stop margin content in a PDF from being pushed around?

### DISCUSSION THREAD

**Discussion by sasja-san** [quarto-dev/quarto-cli#8817] [Q&A]
### Description

Here's what I'm getting. It's the page the furthest to the right which I find the most annoying.

![quarto-margin-push-1](https://github.com/quarto-dev/quarto-cli/assets/712688/80e540a8-3e94-4e86-93c3-d464bc9b4874)

Is there a way for margin content to appear at the top of a page before any other stuff, while at the same time not pushing around anything in the main column?

```qmd
---
keep-tex: true

reference-location: margin

format:
    pdf:
      papersize: a6
      geometry:
        - showframe
---

# Reference page

In a doc with that has a bunch of things in the margin, if 
that should matter.^[Maybe it does, maybe it doesn't...]

{{< pagebreak >}}






# Margin stuff

::::::{.column-margin}
:::{.callout-warning}
This box is placed *after* the heading, before other text.
:::
::::::

This text is written after the content in the margin.

{{< pagebreak >}}






::::::{.column-margin}
:::{.callout-warning}
Be careful at the top of the page!
:::
::::::

# Margin first

Pushing me around!

```

Here's the generated LaTeX. I can't see any obvious bad guys here. Anyone know any tricks? Maybe a `tcolorbox` hack could be done?

```latex

\newpage{}

\marginnote{\begin{footnotesize}

\begin{tcolorbox}[enhanced jigsaw, titlerule=0mm, rightrule=.15mm, colbacktitle=quarto-callout-warning-color!10!white, left=2mm, arc=.35mm, coltitle=black, colback=white, breakable, bottomtitle=1mm, title=\textcolor{quarto-callout-warning-color}{\faExclamationTriangle}\hspace{0.5em}{Warning}, toptitle=1mm, bottomrule=.15mm, toprule=.15mm, opacitybacktitle=0.6, leftrule=.75mm, opacityback=0, colframe=quarto-callout-warning-color-frame]

Be careful at the top of the page!

\end{tcolorbox}

\end{footnotesize}}

\section{Margin first}\label{margin-first}

Pushing me around!

 ```
 
 Using the latest quarto `main` branch.
 
 Any advice on what could be done is welcome.

**Comment by mcanouil** [comment]
Welcome to LaTeX floats 😅
You'll have better luck on more LaTeX dedicated platforms.

Quarto/Pandoc are simply using LaTeX in all its beauty.

**Reply by sasja-san** [reply]
Is there no way to declare

> Oh btw, the stuff that's in this box is actuall 0x0 in size, for any calculations you might need to do...

**Reply by mcanouil** [reply]
You can use raw LaTeX: `` ```{=latex} ``
